File: 118/solution.py
class Solution:
    def generate(self, numRows: int) -> List[List[int]]:
        soln = [[1], [1, 1]]
        if numRows < 3:
            return soln[:numRows]
        prev = soln[-1]
        for row in range(2, numRows):
            new_row = [1]
            for idx in range(1, row):
                new_row.append(prev[idx - 1] + prev[idx])
            new_row.append(1)
            soln.append(new_row)
            prev = new_row
        return soln

File: 118/Solution.java
class Solution {
    public List<List<Integer>> generate(int numRows) {
        List<List<Integer>> soln = new ArrayList<>();
        ArrayList<Integer> temp = new ArrayList<>();
        List<Integer> prev = new ArrayList<>();

        for (int row = 0; row < numRows; row++) {
            temp = new ArrayList<>();
            temp.add(1);

            for (int idx = 1; idx < row; idx++) {
                int val = prev.get(idx - 1) + prev.get(idx);
                temp.add(val);
            }
            if (row > 0) {
                temp.add(1);
            }

            soln.add(temp);
            prev.clear();
            prev.addAll(temp);
        }
        return soln;
    }
}